spec/index.bs

```diff
@@ -1555,31 +1574,32 @@ interface GPUTextureUsage {
       1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be nonzero.
       1. If |descriptor|.{{GPUTextureDescriptor/dimension}} is "1d":
 
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to the owning {{GPUDevice}}'s {{GPULimits/maxTextureSize}}.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to the owning {{GPUDevice}}'s {{GPULimits/maxTextureLayers}}.
         1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be 1.
         1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
 
       1. Else if |descriptor|.{{GPUTextureDescriptor/dimension}} is "2d":
 
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 8192.
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to the owning {{GPUDevice}}'s {{GPULimits/maxTextureSize}}.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to the owning {{GPUDevice}}'s {{GPULimits/maxTextureSize}}.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to the owning {{GPUDevice}}'s {{GPULimits/maxTextureLayers}}.
 
       1. Else (|descriptor|.{{GPUTextureDescriptor/dimension}} is "3d"):
 
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 2048.
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
-        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to the owning {{GPUDevice}}'s {{GPULimits/maxTextureLayers}}.
```

Comment:
For 3D textures I think each dimension should use `maxTextureSize` instead of `maxTextureLayers`, or maybe we could have split limits for 1D, 2D and 3D (3D's default limit can't be more than 2048 because that's what the majority of Vulkan devices support)

Reply:
It seems that we have to split here:
- `maxTextureSize1D = 8192`
- `maxTextureSize2D = 8192`
- `maxTextureSize3D = 2048`
- `maxTextureLayers = 256`